File: 2025-1/sesion-de-clase-19/Persona.cpp
/* 
 * File:   Persona.cpp
 * Author: erichuiza
 * 
 * Created on May 26, 2025, 4:29 PM
 */

#include "Persona.h"

Persona::Persona(int dni, const string nombre) : Registro() {
    this->dni = dni;
    this->setNombre(nombre);
}

//Persona::Persona(const Persona& orig) : Registro(orig) {
//    *this = orig;
//}

int Persona::getDni() const {
    return dni;
}

void Persona::setDni(int dni) {
    this->dni = dni;
}

string Persona::getNombre() const {
    return this->nombre;
}

void Persona::setNombre(string nombre) {
    this->nombre = nombre;
}

bool Persona::cargarDatos(ifstream& archivo) {
    int dni;
    if (!archivo.eof() && archivo >> dni) {
        archivo.ignore();
        getline(archivo, this->nombre, ',');
        
        this->setDni(dni);
        this->setNombre(nombre);
        
        return true;
    }

    return false;
}

void Persona::imprimir(ostream& os) const {
    os << left << setw(15) << this->dni
        << setw(50) << this->nombre;
}

File: 2025-2/sesion-de-clase-26-1/src/Persona.java

import java.util.Scanner;

/**
 *
 * @author erichuiza
 */
public abstract class Persona extends Registro {
    private int dni;
    private String nombre;
    
    public Persona() {
    }
    
    public Persona(int dni, String nombre) {
        this.dni = dni;
        this.nombre = nombre;
    }
    
    public int getDni() {
        return dni;
    }

    public void setDni(int dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    @Override
    protected void cargarDatos(Scanner scanner) {
        this.dni = scanner.nextInt();
        this.nombre = scanner.next();
    }

    @Override
    protected void imprimir() {
        System.out.println("DNI: " + this.dni);
        System.out.println("Nombre: " + this.nombre);
    }    
}
